pkg/operator/ceph/cluster/watcher.go

```diff
@@ -546,9 +551,54 @@ func (c *clientCluster) createNetworkFence(ctx context.Context, pv corev1.Persis
 	return nil
 }
 
-func (c *clientCluster) unfenceAndDeleteNetworkFence(ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster) error {
+func (c *clientCluster) unfenceAndDeleteRbdNetworkFence(ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster) error {
+	networkFence := &addonsv1alpha1.NetworkFence{}
+	err := c.client.Get(ctx, types.NamespacedName{Name: node.Name + "rbd", Namespace: cluster.Namespace}, networkFence)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	} else if errors.IsNotFound(err) {
+		return nil
+	}
+	logger.Infof("node %s does not have taint %s, unfencing networkFence CR", node.Name, corev1.TaintNodeOutOfService)
+
+	// Unfencing is required to unblock the node and then delete the network fence CR
+	networkFence.Spec.FenceState = addonsv1alpha1.Unfenced
+	err = c.client.Update(ctx, networkFence)
+	if err != nil {
+		logger.Errorf("failed to unFence network fence CR. %v", err)
+		return err
+	}
+
+	err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, true, func(ctx context.Context) (bool, error) {
+		err = c.client.Get(ctx, types.NamespacedName{Name: node.Name + "rbd", Namespace: cluster.Namespace}, networkFence)
+		if err != nil && !errors.IsNotFound(err) {
+			return false, err
+		}
+
+		if networkFence.Spec.FenceState != addonsv1alpha1.Unfenced {
+			logger.Infof("waiting for network fence CR %s to get in %s state before deletion", networkFence.Name, addonsv1alpha1.Unfenced)
+			return false, err
+		}
+
+		logger.Infof("successfully unfenced rbd network fence cr %q, proceeding with deletion", networkFence.Name)
+
+		err = c.client.Delete(ctx, networkFence)
+		if err == nil || errors.IsNotFound(err) {
+			logger.Infof("successfully deleted network fence CR %s", networkFence.Name)
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return pkgerror.Wrapf(err, "timeout out deleting the rbd network fence CR %s", networkFence.Name)
+	}
+
+	return nil
+}
+
+func (c *clientCluster) unfenceAndDeleteCephFSNetworkFence(ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster) error {
```

Comment:
Can we have a common helper method for unfencing? If the only difference between `unfenceAndDeleteCephFSNetworkFence()` and `unfenceAndDeleteRbdNetworkFence()` is the name of the CR and the log messages, then we can make the name a parameters and keep a single method.

Reply:
Yes, made the changes.